src/components/map/Map.tsx

```diff
@@ -50,61 +51,51 @@ const MapIcon = ({
   return <MarkerIcon color={iconColor} size={iconSize} />;
 };
 
-type TCluster = {
-  clusterColor: string;
-  geometry: { coordinates: [number, number] };
-  id: string;
-  onPress: () => void;
-  properties: { point_count: number };
-};
-
-const renderCluster = (cluster: TCluster) => {
+const renderCluster = (cluster) => {
   const { clusterColor: backgroundColor, geometry, id, onPress, properties = {} } = cluster;
   const { point_count: points } = properties;
 
   return (
-    <Marker
+    <MapLibreGL.PointAnnotation
       key={`cluster-${id}`}
-      coordinate={{
-        longitude: geometry.coordinates[0],
-        latitude: geometry.coordinates[1]
-      }}
-      style={styles.clusterMarker}
-      onPress={onPress}
+      id={`cluster-${id}`}
```

Comment:
what is the `id` for?

Reply:
The id attribute in the <MapLibreGL.PointAnnotation> component is used to uniquely identify each annotation (point or cluster) on the map. This unique identifier is crucial for various reasons:

1. **Interaction Handling**: The id allows you to handle user interactions with specific annotations, such as selecting, deselecting, or clicking on a point. When an annotation is selected, the map can identify which annotation was interacted with based on its id.

2. **Performance Optimization**: By using unique identifiers, the map library can efficiently manage and update only the annotations that have changed, rather than re-rendering all annotations. This can lead to performance improvements, especially when dealing with a large number of points.

3. **Data Association**: The id helps in associating the annotations with the underlying data points. For example, if you need to display additional information about a specific location when it is clicked, you can use the id to fetch the relevant data.